src/mame/epoch/cassvisn.cpp

```diff
@@ -80,12 +79,31 @@ INPUT_PORTS_END
 
 DEVICE_IMAGE_LOAD_MEMBER(cassvisn_state::cart_load)
 {
+	if (!image.loaded_through_softlist())
+		return std::make_pair(image_error::UNSUPPORTED, "Cartridges must be loaded from the software list");
+
 	uint32_t size = m_cart->common_get_size("prg");
+
+	if (size != 0xf00)
+		return std::make_pair(image_error::UNSUPPORTED, "prg region size must be 0xf00 in size");
+
 	m_cart->rom_alloc(size, GENERIC_ROM16_WIDTH, ENDIANNESS_LITTLE);
 	m_cart->common_load_rom(m_cart->get_rom_base(), size, "prg");
-	memcpy(memregion("maincpu")->base(), m_cart->get_rom_base(), size);
+	uint8_t* prgbase = memregion("maincpu")->base();
+	memcpy(prgbase, m_cart->get_rom_base(), size);
+
+	for (int i = 0; i < size; i += 2)
+	{
+		uint8_t temp = prgbase[i + 0];
+		prgbase[i + 0] = prgbase[i + 1];
+		prgbase[i + 1] = temp;
+	}
```

Comment:
Or is it just a matter of changing the cpu device endianness?

Reply:
is Endianess really a thing in this case? it's a single 12 bit byte?  I find this a little confusing.

Reply:
Here it'd just be for letting MAME know the byte order for each ROM word. TMS0980 and HMCS40 are similar, with 9-bit and 12-bit ROM words.

Reply:
I changed everything to 'big' and the swap is still needed, I must be missing something here.

Reply:
I see, that's weird. Just add a TODO note then.